### PR TITLE
Fix triggers for ILLINK job

### PR DESCRIPTION
**Pull request by swaroop-sridhar**
This PR has two checkins: 
1) Fix the trigger phrase for ILLink jobs so that PRs can request ILLink runs per architecture.
2) Fix the way ILLink.exe is passed to runtest.cmd by the lab script.

### CODE REVIEW

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

**Comment by swaroop-sridhar** [comment]
@jashook please review

**Comment by swaroop-sridhar** [comment]
I'm trying to fix the PR triggers currently reading:
`@dotnet-bot test illink Windows_NT Checked illink`
to:
`@dotnet-bot test Windows_NT X64 Checked illink`
`@dotnet-bot test Windows_NT X86 Checked illink`
etc.

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

File: netci.groovy [hunk 1]
```diff
@@ -325,10 +325,6 @@ def static getJobName(def configuration, def architecture, def os, def scenario,
                 // we don't care about the configuration for the formatting job. It runs all configs
                 baseName = architecture.toLowerCase() + '_' + os.toLowerCase()
             }
-            else if (scenario == 'illink') {
```
Comment (swaroop-sridhar) [inline netci.groovy]: The scenario name is already appended.

File: netci.groovy [hunk 2]
```diff
@@ -1260,7 +1256,7 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
                        "(?i).*test\\W+${os}\\W+${architecture}\\W+${scenario}.*")
                     break
                 case 'illink':
-                    Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} via ILLink", "(?i).*test\\W+illink\\W+${os}\\W+${configuration}\\W+${scenario}.*")
+                    Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} via ILLink", "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}\\W+${scenario}.*")
```
Comment (jashook) [inline netci.groovy]: I'm guessing you want illink somewhere in teh trigger phrase

Comment (swaroop-sridhar) [inline netci.groovy]: We get ILLINK because `${scenario}` is included in the trigger phrase

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

File: netci.groovy (hunk 2 above)
Comment (jashook) [inline netci.groovy]: gotcha

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

**Comment by swaroop-sridhar** [comment]
@dotnet-bot  test this please

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test CI Please

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test this please
@dotnet-bot test CI Please

**Pull request merged**